include/cassandra.h

```diff
@@ -1155,6 +1155,37 @@ cass_cluster_set_tcp_keepalive(CassCluster* cluster,
                                cass_bool_t enabled,
                                unsigned delay_secs);
 
+/**
+ * Sets the amount of time between heartbeat messages and controls the amount
+ * of time the connection must be idle before sending heartbeat messages. This
+ * is useful for preventing intermediate network devices from dropping
+ * connections.
+ *
+ * <b>Default:</b> 30 seconds
+ *
+ * @public @memberof CassCluster
+ *
+ * @param[in] cluster
+ * @param[in] interval_ms Use 0 to disable heartbeat messages
+ */
+CASS_EXPORT void
+cass_cluster_set_connection_heartbeat_interval(CassCluster* cluster,
+                                               unsigned interval_secs);
+
+/**
+ * TODO
+ *
+ * <b>Default:</b> 60 seconds
+ *
+ * @public @memberof CassCluster
+ *
+ * @param[in] cluster
+ * @param[in] interval_ms
+ */
+CASS_EXPORT void
+cass_cluster_set_connection_idle_interval(CassCluster* cluster,
```

Comment:
curious why there needs to be two settings here. I think of heartbeat interval as "send a request if the connection is idle for x seconds".